### PR TITLE
Notification type for recovery

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java [hunk 1]
```diff
@@ -173,43 +173,51 @@ private boolean isFewBlocksAfterInitialization(Task task) {
             String enclaveChallenge = smsService.getEnclaveChallenge(chainTaskId, task.isTeeNeeded());
             if (task.isTeeNeeded() && enclaveChallenge.isEmpty()) continue;
 
-            Optional<RecoveryAction> oRecoveryAction = getAppropriateRecoveryAction(task, replicate, blockNumber);
-            if (!oRecoveryAction.isPresent()) continue;
+            Optional<TaskNotificationType> optionalTaskNotificationType = getAppropriateRecoveryAction(task, replicate, blockNumber);
+            if (!optionalTaskNotificationType.isPresent()) continue;
 
             // generate contribution authorization
             ContributionAuthorization authorization = signatureService.createAuthorization(
```
Comment (zied-guesmi) [inline src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java]: Shouldn't we generate the contribution authorization only in the case of please_contribute notification?

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java [hunk 2]
```diff
@@ -173,43 +173,51 @@ private boolean isFewBlocksAfterInitialization(Task task) {
...
                     walletAddress, chainTaskId, enclaveChallenge);
 
-            InterruptedReplicateModel interruptedReplicate = InterruptedReplicateModel.builder()
+            //TODO Send only required extras
+            TaskNotificationExtra taskNotificationExtra = TaskNotificationExtra.builder()
                     .contributionAuthorization(authorization)
-                    .recoveryAction(oRecoveryAction.get())
+                    .blockNumber(task.getConsensusReachedBlockNumber())
```
Comment (zied-guesmi) [inline src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java]: We shouldn't send the blockNumber when the notif is pleas_contribute (consensus is not yet reached)

File: src/main/java/com/iexec/core/task/listener/TaskListeners.java [hunk 3]
```diff
@@ -83,7 +84,7 @@ public void onTaskConsensusReached(ConsensusReachedEvent event) {
             notificationService.sendTaskNotification(TaskNotification.builder()
                     .taskNotificationType(TaskNotificationType.PLEASE_REVEAL)
                     .chainTaskId(chainTaskId)
-                    .blockNumber(event.getBlockNumber())
+                    .taskNotificationExtra(TaskNotificationExtra.builder().blockNumber(event.getBlockNumber()).build())
```
Comment (zied-guesmi) [inline src/main/java/com/iexec/core/task/listener/TaskListeners.java]: Why not just TaskNotificationExtra(event.getBlockNumber())

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java (hunk 1 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java]: Didn't want to do it (-> https://github.com/iExecBlockchainComputing/iexec-core/blob/dc24414a9426e1c0284b57ac2ae4197f6581f1d2/src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java#L183) but done now :P

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java (hunk 2 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java]: Done

File: src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java (hunk 1 above)
Comment (zied-guesmi) [reply src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java]: Ah sorry didn't understand you meant that, thanks!

**Pull request merged**